# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/dom_localization/test_translateElements.html
# Commit: 85194e5b7a9e
# Full Hash: 85194e5b7a9e6ed9114ee46aef0b5efb819cca9b
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-04 03:56:33
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 181487bdf6c0 -r 85194e5b7a9e dom/l10n/tests/mochitest/dom_localization/test_translateElements.html
--- a/dom/l10n/tests/mochitest/dom_localization/test_translateElements.html	Tue Aug 03 16:25:13 2021 +0000
+++ b/dom/l10n/tests/mochitest/dom_localization/test_translateElements.html	Tue Aug 03 16:25:13 2021 +0000
@@ -7,20 +7,25 @@
   <link rel="stylesheet" type="text/css" href="chrome://mochikit/content/tests/SimpleTest/test.css">
   <script type="application/javascript">
   "use strict";
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource("title = Hello World"));
-    bundle.addResource(new FluentResource("link =\n    .title = Click me"));
-    yield bundle;
-  }
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
+title = Hello World
+link =
+    .title = Click me
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   window.onload = async function() {
     SimpleTest.waitForExplicitFinish();
 
     const domLoc = new DOMLocalization(
-      [],
+      ["/mock.ftl"],
       false,
-      { generateBundles },
+      l10nReg,
+      ["en-US"],
     );
 
     const p1 = document.querySelectorAll("p")[0];